.github/workflows/test.yml

```diff
@@ -95,30 +104,29 @@ jobs:
           sudo cp -f ../../go-controller/_output/go/bin/ovn* .
           echo "ref: $(git rev-parse  --symbolic-full-name HEAD)  commit: $(git rev-parse  HEAD)" > git_info
           docker build -t ovn-daemonset-f:dev -f Dockerfile.fedora .
-          mkdir _output
-          docker save ovn-daemonset-f:dev > _output/${CI_IMAGE_MASTER_TAR}
-        popd
+        popd 
 
     - name: Cache master image
-      if: steps.is_master_image_build_needed.outputs.MASTER_IMAGE_RESTORED != 'true' && success()
-      continue-on-error: true
+      if: (steps.is_master_image_build_needed.outputs.LOAD_MASTER_IMAGE_TO_CACHE == 'true' || steps.is_master_image_build_needed.outputs.MASTER_IMAGE_RESTORED != 'true') && success()
```

Comment:
nit continued:
and then here we would check for `MASTER_IMAGE_RESTORED==true && MASTER_IMAGE_RESTORED_FROM_CACHE==false`
It just all reads more natural this way.